project/.gitlab-ci.yml

```diff
@@ -59,3 +66,27 @@ python3.7:
 python3.8:
   <<: *test
   image: python:3.8
+
+python3.9:
+  <<: *test
+  image: python:3.9
+
+
+pages:
+  stage: deploy
+  <<: *install-deps
+  image: python:3.8
+  dependencies:
+  - python3.6
+  - python3.7
+  - python3.8
+  - python3.9
+  script:
+  - poetry run duty coverage
+  - poetry run duty docs
```

Comment:
Retrieve artifacts from the listed jobs: the `.coverage-*` files. Combine them, generate the HTML report, build the docs.